lib/spack/spack/test/stage.py

```diff
@@ -177,82 +194,32 @@ def get_stage_path(stage, stage_name):
 
 
 @pytest.fixture
-def bad_stage_path():
-    """Temporarily ensure there is no accessible path for staging."""
-    current = spack.config.get('config:build_stage')
-    spack.config.set('config', {'build_stage': '/no/such/path'}, scope='user')
-    yield
-    spack.config.set('config', {'build_stage': current}, scope='user')
-
-
-@pytest.fixture
-def non_user_path_for_stage(monkeypatch):
-    """Temporarily use a Linux-standard non-user path for staging. """
-    def _can_access(path, perms):
-        return True
-
-    current = spack.config.get('config:build_stage')
-    spack.config.set('config', {'build_stage': [_non_user_root]}, scope='user')
-    monkeypatch.setattr(os, 'access', _can_access)
-    yield
-    spack.config.set('config', {'build_stage': current}, scope='user')
-
-
-@pytest.fixture
-def instance_path_for_stage():
-    """
-    Temporarily use the "traditional" spack instance stage path for staging.
-
-    Note that it can be important for other tests that the previous settings be
-    restored when the test case is over.
-    """
-    current = spack.config.get('config:build_stage')
+@pytest.mark.nomockstage
+def instance_path_for_stage(clear_stage_root):
+    """Use the spack instance stage path for the stage root."""
     base = canonicalize_path(os.path.join('$spack', 'test-stage'))
     mkdirp(base)
-    path = tempfile.mkdtemp(dir=base)
-    spack.config.set('config', {'build_stage': path}, scope='user')
-    yield
-    spack.config.set('config', {'build_stage': current}, scope='user')
-    shutil.rmtree(base)
+    test_path = tempfile.mkdtemp(dir=base)
 
+    with spack.config.override('config:build_stage',  test_path):
+        yield spack.stage.get_stage_root()
 
-@pytest.fixture
-def tmp_path_for_stage(tmpdir):
-    """
-    Use a temporary test directory for staging.
-
-    Note that it can be important for other tests that the previous settings be
-    restored when the test case is over.
-    """
-    current = spack.config.get('config:build_stage')
-    spack.config.set('config', {'build_stage': [str(tmpdir)]}, scope='user')
-    yield tmpdir
-    spack.config.set('config', {'build_stage': current}, scope='user')
+    shutil.rmtree(base)
```

Comment:
What if someone already happens to have a file in `$spack` named `test-stage`?

Reply:
Fair enough.

How about I make it a hidden file?  Or I could sequence it. <grin>

Reply:
Of the two options I mentioned, which would you prefer?  Or do you have an alternate?

Reply:
FYI. Changed to hidden directory.

Reply:
That at least makes it more likely it's safe. We should probably put `spack` in the name to be a little safer still

Reply:
Done